wicket-core/src/main/java/org/apache/wicket/resource/FileSystemResource.java

```diff
@@ -0,0 +1,145 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.resource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import org.apache.wicket.Application;
+import org.apache.wicket.WicketRuntimeException;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.apache.wicket.request.resource.AbstractResource;
+import org.apache.wicket.request.resource.PartWriterCallback;
+import org.apache.wicket.util.lang.Args;
+
+/**
+ * Used to provide resources based on the on Java NIO FileSystem API.<br>
+ * <br>
+ * For more information see {@link org.apache.wicket.markup.html.media.FileSystemResourceReference}
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class FileSystemResource extends AbstractResource
+{
+	private static final long serialVersionUID = 1L;
+
+	private final Path path;
+
+	/**
+	 * Creates a new file system resource based on the given path
+	 * 
+	 * @param path
+	 *            the path to be read for the resource
+	 */
+	public FileSystemResource(Path path)
+	{
+		Args.notNull(path, "path");
+		this.path = path;
+	}
+
+	/**
+	 * Creates a new resource response and reads the given path
+	 */
+	@Override
+	protected ResourceResponse newResourceResponse(Attributes attributes)
+	{
+		try
+		{
+			long size = getSize();
+			ResourceResponse resourceResponse = new ResourceResponse();
+			resourceResponse.setContentType(getMimeType());
+			resourceResponse.setAcceptRange(ContentRangeType.BYTES);
+			resourceResponse.setContentLength(size);
+			resourceResponse = configureResourceResponse(resourceResponse);
+			RequestCycle cycle = RequestCycle.get();
+			Long startbyte = cycle.getMetaData(CONTENT_RANGE_STARTBYTE);
+			Long endbyte = cycle.getMetaData(CONTENT_RANGE_ENDBYTE);
+			resourceResponse.setWriteCallback(new PartWriterCallback(getInputStream(), size,
+				startbyte, endbyte));
+			return resourceResponse;
+		}
+		catch (IOException e)
+		{
+			throw new WicketRuntimeException(
+				"An error occured while processing the media resource response", e);
+		}
+	}
+
+	/**
+	 * Configures the resource response by overriding it with additional information
+	 * 
+	 * @param resourceResponse
+	 *            the resource response to be configured
+	 * 
+	 * @return the resource response to be configured
+	 */
+	protected ResourceResponse configureResourceResponse(ResourceResponse resourceResponse)
```

Comment:
I think there is no need of `#configureResourceResponse()`. 
`org.apache.wicket.request.resource.AbstractResource#setResponseHeaders